src/sources/static_metrics.rs

```diff
@@ -0,0 +1,312 @@
+use std::collections::BTreeMap;
+use std::num::NonZero;
+use std::time::Duration;
+
+use chrono::Utc;
+use futures::StreamExt;
+use serde_with::serde_as;
+use tokio::time;
+use tokio_stream::wrappers::IntervalStream;
+use vector_lib::configurable::configurable_component;
+use vector_lib::internal_event::{CountByteSize, InternalEventHandle as _};
+use vector_lib::{config::LogNamespace, ByteSizeOf, EstimatedJsonEncodedSizeOf};
+
+use crate::{
+    config::{SourceConfig, SourceContext, SourceOutput},
+    event::{
+        metric::{MetricData, MetricName, MetricSeries, MetricTime, MetricValue},
+        EventMetadata, Metric, MetricKind,
+    },
+    internal_events::{EventsReceived, StaticMetricsBytesReceived, StreamClosedError},
+    shutdown::ShutdownSignal,
+    SourceSender,
+};
+
+/// Configuration for the `static_metrics` source.
+#[serde_as]
+#[configurable_component(source(
+    "static_metrics",
+    "Produce static metrics defined in configuration."
+))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct StaticMetricsConfig {
+    /// The interval between metric emitting, in seconds.
+    #[serde_as(as = "serde_with::DurationSecondsWithFrac<f64>")]
+    #[serde(default = "default_interval")]
+    #[configurable(metadata(docs::human_name = "Emitting interval"))]
+    pub interval_secs: Duration,
+
+    /// Overrides the default namespace for the metrics emitted by the source.
+    #[serde(default = "default_namespace")]
+    pub namespace: String,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    pub metrics: Vec<StaticMetricConfig>,
+}
+
+impl Default for StaticMetricsConfig {
+    fn default() -> Self {
+        Self {
+            interval_secs: default_interval(),
+            metrics: Vec::default(),
+            namespace: default_namespace(),
+        }
+    }
+}
+
+/// Tag configuration for the `internal_metrics` source.
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct StaticMetricConfig {
+    /// Name of the static metric
+    pub name: String,
+
+    /// "Observed" value of the static metric
+    pub value: MetricValue,
+
+    /// Kind of the static metric - either absolute or incremental
+    #[serde(default = "default_metric_kind")]
+    pub kind: MetricKind,
+
+    /// Key-value pairs representing tags and their values to add to the metric.
+    #[configurable(metadata(
+        docs::additional_props_description = "An individual tag - value pair."
+    ))]
+    pub tags: BTreeMap<String, String>,
+}
+
+fn default_interval() -> Duration {
+    Duration::from_secs_f64(1.0)
+}
+
+fn default_namespace() -> String {
+    "vector".to_owned()
```

Comment:
I'm not sure about defaulting this to `vector`. We currently use the `vector` namespace for internal metrics. These would be user-configured metrics. I realize the use-case motivating this PR would like to put them in `vector`, but I'd suggest we make the default something like `static` to avoid unintentional conflicts. What would you think?

Reply:
That makes sense actually. I didn't give it much thought while working on this PR and I just re-used the default value from internal metrics. I will update it to `static`, it can be easily changed in configuration.